### PR TITLE
[TT-5185] Fix for grpc streams not getting flushed immediately

### PR DESCRIPTION
**Pull request by JayShahSTL**
Ref: https://github.com/TykTechnologies/tyk/issues/4020

Implement missing FLush() for the customResponseWriter.

## Description

Implement missing FLush() for the customResponseWriter to be used by go grpc package.

## Related Issue

https://github.com/TykTechnologies/tyk/issues/4020

## Motivation and Context

Streaming without buffering.
## How This Has Been Tested

With custom plugins. 

## Types of changes

- [ x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Refactoring or add test (improvements in base code or adds test coverage to functionality)

## Checklist

- [ x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side). If pulling from your own
      fork, don't request your `master`!
- [ x] Make sure you are making a pull request against the **`master` branch** (left side). Also, you should start
      *your branch* off *our latest `master`*.
- [ ] My change requires a change to the documentation.
  - [ ] If you've changed APIs, describe what needs to be updated in the documentation.
  - [ ] If new config option added, ensure that it can be set via ENV variable
- [ ] I have updated the documentation accordingly.
- [ ] Modules and vendor dependencies have been updated; run `go mod tidy && go mod vendor`
- [ ] When updating library version must provide reason/explanation for this update.
- [ ] I have added tests to cover my changes.
- [ x] All new and existing tests passed.
- [ x] Check your code additions will not fail linting checks:
  - [ ] `go fmt -s`
  - [ ] `go vet`

### CODE REVIEW

**Comment by JayShahSTL** [comment]
> I think this may be a problematic change for some people that use custom response writers, or wrap them incompletely (as per godoc). Please add error handling that confirms implementation.

File: gateway/mw_go_plugin.go [hunk 1]
```diff
@@ -63,7 +63,9 @@ func (w *customResponseWriter) WriteHeader(statusCode int) {
 }
 
 func (w *customResponseWriter) Flush() {
-        w.ResponseWriter.(http.Flusher).Flush()
+       if _, ok := w.ResponseWriter.(http.Flusher); ok {
```
Comment (titpetric) [inline gateway/mw_go_plugin.go:66]: I feel like you can do `flusher` instead of `_` and then just use flusher.Flush() without a double cast. :pray:

Comment (JayShahSTL) [reply gateway/mw_go_plugin.go]: Hi @titpetric Can this be approved please?

Comment (titpetric) [reply gateway/mw_go_plugin.go]: I need to do some internal housekeeping to include it in the next sprint. It's gonna be some delay, but unless some new concerns rise up, it's a merge candidate. 👍

Comment (JayShahSTL) [reply gateway/mw_go_plugin.go]: Thank you!

**Comment by titpetric** [comment]
Closing this, change has been added as part of #4299, will be included in 4.0.9 release - please feedback if you need anything else. I'd personally be very interested why you're driving gRPC over the go plugins... @JayShahSTL ;)

**Pull request closed without merging**